internal-services/catalog/publish-index-image-task.yaml

```diff
@@ -49,14 +49,53 @@ spec:
         quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
       script: |
         #!/usr/bin/env bash
+        set -euo pipefail
+
         PATH=/bin:/usr/bin:/usr/local/bin
         export PATH
 
-        # do not authenticate if the source is redhat's "registry-proxy" which is unauthenticated.
+        AUTH_PARAM=()
+        # Do not authenticate if the source is Red Hat's "registry-proxy" which is unauthenticated.
         if [[ ! "$(params.sourceIndex)" =~ ^registry-proxy(\-stage)?.engineering.redhat.com ]]; then
             AUTH_PARAM=("--src-creds" "${SOURCE_INDEX_CREDENTIAL}")
         fi
+        TARGET_AUTH="--dest-creds ${TARGET_INDEX_CREDENTIAL}"
+
+        # Check if the target image exists
+        echo "Checking if target image exists: $(params.targetIndex)"
+        set +e
+        TARGET_DIGEST=$(skopeo inspect \
+            --tls-verify=false \
+            "${TARGET_AUTH}" \
+            "docker://$(params.targetIndex)" \
+            --format '{{.Digest}}' 2>/dev/null)
+        INSPECT_EXIT_CODE=$?
+        set -e
+
+        # Get the source image digest
+        echo "Getting source image digest: $(params.sourceIndex)"
+        SOURCE_DIGEST=$(skopeo inspect \
+            --tls-verify=false \
+            "${AUTH_PARAM[@]}" \
+            "docker://$(params.sourceIndex)" \
+            --format '{{.Digest}}')
+
+        if [ $INSPECT_EXIT_CODE -eq 0 ]; then
+            if [ "$SOURCE_DIGEST" == "$TARGET_DIGEST" ]; then
+                echo "Image already exists in target registry with the same digest." \
+                     "Skipping copy." | tee "$(results.requestMessage.path)"
+                exit 0
+            else
+                echo "Image exists in target registry but digests do not match." \
```

Comment:
The targetIndex is the catalog and will always exist - except for hotfix, which is always different.
So only checking if the digest is the same is enough.

Reply:
Ah got it, fixed now, as you suggested. PTAL!